### PR TITLE
Added upload_folder and donwload_folder scripts

### PR DESCRIPTION
**Pull request by leocosma**
I added two scripts which allow to download and upload folders recursively.

Authentication credential are requested during the execution, or are retrieved from environment variables.

### CODE REVIEW

File: clfpy/tests/upload_folder [hunk 1]
```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+
+"""
+Script for uploading an entire folder
+
+Usage: update_folder FOLDER DESTINATION
```
Comment (rojsc) [inline clfpy/tests/upload_folder]: Should be "Usage: upload_folder"

File: clfpy/tests/upload_folder [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+  FOLDER       the folder to be uploaded, absolute or relative path
+  DESTINATION  the gss destination, which must exist before  (e.g. it4i_anselm://
```
Comment (rojsc) [inline clfpy/tests/upload_folder]: Please add either 'home' or 'scratch' to the GSS URIs.

**Comment by rojsc** [comment]
I'm not quite sure what to make of this pull request.
It adds files to the "tests" folder, but actually implements new functionality. At the same time, the scripts don't extend the actual clfpy library, so they will be difficult to use for others. A better approach would have been to implement the two methods directly in the clfpy.GssClient class.

**Review by rojsc — COMMENTED** [review]
Both files need ".py" filename extensions.

**Comment by leocosma** [comment]
Understood. I'll try to implement the methods in the clfpy.GssClient class.

**Comment by leocosma** [comment]
I'vemoved the new GSS functions in the GSS Client implementation and modified the files in the TESTS folder. I think last commits are better.

**Comment by rojsc** [comment]
That looks much better, thank you!
Please remove the bits marked as workarounds (path insertions), then I can merge.

**Comment by leocosma** [comment]
Done.

**Comment by rojsc** [comment]
Merged.
Thank you for your contribution, @leocosma!

**Comment by leocosma** [comment]
You're welcome. When will it be possible to update the clfpy package from python installation?

**Comment by rojsc** [comment]
Good point. I updated the pypi package, you should be able to upgrade with pip install --upgrade clfpy.

**Pull request merged**